### PR TITLE
feature(unlock-app): Update General settings for an event collection

### PR DESCRIPTION
**Pull request by 0xTxbi**
# Description
This PR introduces the ability to update the general settings for an event collection.

## ScreenGrab
Screenshot 2024-09-24 at 15 55 30

## Flow
[Loom](https://www.loom.com/share/45b27bb5bbcc417a8ec36a521b10bd8e?sid=426e1ad8-b2fb-4987-8afc-3cb225cad08c)

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/content/events-collection/EventCollectionSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  eventCollection: EventCollection
+}
+
+export const EventCollectionSettings = ({
+  eventCollection,
+}: EventCollectionSettingsProps) => {
+  const [selectedIndex, setSelectedIndex] = useState(0)
+  const { account } = useAuth()
+
+  const isManager =
```
Comment (julien51) [inline unlock-app/src/components/content/events-collection/EventCollectionSettings.tsx:23]: We probably want to move that to dinner utile file some we're doing it multiple times

**Review by julien51 — APPROVED** [review]

File: unlock-app/src/components/content/events-collection/EventCollectionSettings.tsx (hunk 1 above)
Comment (0xTxbi) [reply unlock-app/src/components/content/events-collection/EventCollectionSettings.tsx:23]: sure thing!

**Comment by 0xTxbi** [comment]
> I am not sure we need both a "save links" and "add liunks" button. Please make sure it saves automatically when adding and/or deleting!

alright!

File: unlock-app/src/components/content/events-collection/settings/General.tsx [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+          {/* Save button for Links */}
+          <div className="flex flex-end w-full pt-8 flex-row-reverse">
+            <Button
+              loading={methods.formState.isSubmitting}
+              type="submit"
+              className="w-48"
+            >
+              Save Links
+            </Button>
+          </div>
```
Comment (julien51) [inline unlock-app/src/components/content/events-collection/settings/General.tsx:253]: This is not needed, let's auto-save when links are added or removed.

**Review by julien51 — CHANGES_REQUESTED** [review]
One quick change/fix

File: unlock-app/src/components/content/events-collection/settings/General.tsx (hunk 2 above)
Comment (0xTxbi) [reply unlock-app/src/components/content/events-collection/settings/General.tsx]: done!

**Review by julien51 — APPROVED** [review]

**Pull request merged**